docker-style/action.yml

```diff
@@ -0,0 +1,80 @@
+name: >
+  Lint Dockerfile
+
+description: >
+  Evaluate the quality of your project Dockerfile(s) by using `hadolint
+  <https://github.com/hadolint/hadolint/>`_. This action is expected to be used within a
+  matrix job to lint Dockerfile(s) from multiple directories. The action uses `hadolint-gh-action
+  <https://github.com/jbergstroem/hadolint-gh-action>`_ behind the scenes. If you want to evaluate multiple
+  Dockerfiles contained in various directories of the provided directory, use the recursive option.
+
+  This action emphasis the fact of having Dockerfile(s) contained inside the "docker" directory in the
+  root of the project. Exception can be added as it is the case for the ".devcontainer" directory.
+
+  .. warning::
+      Two checks are performed and failing any of them results in the action failure:
+      - the root of the project must contain a docker directory;
+      - the provided directory must be allowed, i.e. is either "docker" or ".devcontainer".
+
+inputs:
+
+  # Required inputs
+
+  directory:
+    description: >
+      Directory from which to search for Dockerfile(s).
+    required: true
+    type: string
+
+  # Optional inputs
+
+  recursive:
+    description: >
+      Search for Dockerfile(s) recursively.
+    required: false
+    default: false
+    type: bool
+
+  error-level:
+    description: >
+      Fail action based on hadolint output (-1: never, 0: error, 1: warning, 2: info)
+    required: false
+    default: 2
+    type: int
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: "Check docker directory existence"
+      shell: bash
+      run: |
+        if [ ! -d "${{ github.workspace }}/docker" ]; then
+          echo "Directory 'docker' does not exist. Except in special cases, e.g. '.devcontainer', we emphasize that dockerfiles are expected to be in the docker directory in the root of the project."
```

Comment:
We have a logging action that allows for a colorized output. It has three levels: info, warning, and error. It would be great to add it here. Take a look to one of the `doc-deploy-*` actions. Those use it heavily.

Reply:
I'd agree to use this action as well if possible - the output looks nice!